### PR TITLE
rear and front join tweaks

### PR DESCRIPTION
**Pull request by saina-ramyar**
In this PR, platoon identification logic, as well as some logging are added to enable front and rear joins.
# PR Details
## How Has This Been Tested?
Integration tested at TFHRC

## Types of changes

- [x] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: platooning_strategic_IHP/src/platoon_strategic_ihp.cpp [hunk 1]
```diff
@@ -2714,11 +2714,12 @@ namespace platoon_strategic_ihp
...
-                // End the method if time out.
+                // Clean out current plan, and platoon info if the joiner would only be the second member
                 pm_.clearActionPlan();
-                pm_.currentPlatoonID = "";
-                pm_.platoonLeaderID = "";
-                pm_.targetPlatoonID = "";
+                if (pm_.getTotalPlatooningSize() == 2)
+                {
+                    pm_.resetPlatoon();
+                })
```
Comment (saina-ramyar) [inline platooning_strategic_IHP/src/platoon_strategic_ihp.cpp:2722]: looks like an extra parenthesis

File: platooning_strategic_IHP/src/platoon_strategic_ihp.cpp [hunk 2]
```diff
@@ -2232,12 +2240,7 @@ namespace platoon_strategic_ihp
                     ROS_DEBUG_STREAM("Received negative response for plan id = " << pm_.current_plan.planId);
                     // Forget about the previous plan totally
                     pm_.clearActionPlan();
-                    if (pm_.getTotalPlatooningSize() == 1)
-                    {
-                        pm_.currentPlatoonID = "";
-                        pm_.platoonLeaderID = "";
-                        pm_.targetPlatoonID = "";
-                    }
+                    pm_.resetPlatoon();
```
Comment (saina-ramyar) [inline platooning_strategic_IHP/src/platoon_strategic_ihp.cpp:2243]: I think we need a log statement here.

File: platooning_strategic_IHP/src/platoon_strategic_ihp.cpp (hunk 1 above)
Comment (TonysCousin) [reply platooning_strategic_IHP/src/platoon_strategic_ihp.cpp:2722]: Good eye!

File: platooning_strategic_IHP/src/platoon_strategic_ihp.cpp (hunk 2 above)
Comment (TonysCousin) [reply platooning_strategic_IHP/src/platoon_strategic_ihp.cpp:2243]: Added.

**Review by TonysCousin — APPROVED** [review]

**Review by TonysCousin — APPROVED** [review]

**Pull request merged**